pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -421,7 +421,7 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 					},
 					{
 						Alert: "OrphanedVirtualMachineInstances",
-						Expr:  intstr.FromString("((count by (node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'})) or (count by (node)(kube_pod_info{pod=~'virt-launcher.*'})*0)) == 0"),
+						Expr:  intstr.FromString("(sum by(node) (kube_pod_status_ready{condition='true',pod=~'virt-handler.*'}* on(pod) group_left(node) (sum by(node, pod) (kube_pod_info{pod=~'virt-handler.*'}))+ on(node) group_left(_blah) (sum by(node) (kube_pod_info{pod=~'virt-launcher.*', node != ''}*0)) or (sum by(node) (kube_pod_info{pod=~'virt-launcher.*', node != ''})*0))) == 0"),
```

Comment:
Hey,
Maybe i'm missing something but IIUC all we need to do is to replace count with sum:
`count` doesn't consider the value of the metric so if virt-handler isn't ready and such metric exist than `count` will return 1 because it count 1 metric(doesn't care about the value).
`sum` just sum up all the metric's values so it will consider the value of the metric which means that if virt-handler isn't ready sum will return 0 even though this metric exist(because the value of the metric is 0 ).

Why did you replace the all the query can you explain?

Reply:
@Barakmor1 Hi. The result of this query should be a vector `(1,1,0,1,1,0,0..)` each element in this vector should represent a node in a cluster. `0` value indicates one of two things:
* A node with running virt-launcher but w/o a running virt-handler
* A node with running virt-launcher and unhealthy virt-handler

Since we don't care about a node with only virt-handler running, we first calculate a vector that represents nodes with both virt-handler and virt-launcher running on them. Then we union this vector with a vector of nodes with running virt-launchers. This way we get the desired vector, and the comparison `==0` evaluated on each element on the resulted vector

Reply:
what about this query (a bit simpler):

((sum by (node) (kube_pod_status_ready{condition='false',pod=~'virt-handler.*'} * on(pod) group_left(node) kube_pod_info{pod=~'virt-handler.*'}) ) == 1) or (count by (node)( kube_pod_info{pod=~'virt-launcher.*'})*0) == 0